### PR TITLE
chore: updating storybook-react docs

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## Description

This PR improves the organization and documentation of our Storybook react implementation by introducing a more structured approach to component documentation and story organization. The changes make it easier for developers to navigate, understand, and use our design tokens and react components.

### Key Improvements:

- Removed `Text` docs component in favor of `design-system-react` `Text` component
- Consolidated Shadow stories for tailwind classnames and design tokens
- Migrated inline styles `style={{` to tailwind classnames `className="`
- Reorganized stories into logical categories (Getting Started, Design Tokens, React Components, Docs Components)

## Related issues

Fixes: #129

## Manual testing steps

1. Pull this branch
2. Run `yarn cache clean && yarn && yarn build && yarn && yarn storybook` to start Storybook
3. Navigate through the new story hierarchy
4. Verify that all design token stories work
5. Verify component stories work

### Before

https://github.com/user-attachments/assets/e9dbcc8f-b63f-4bd4-961b-4bf62904d9b6

### After

https://github.com/user-attachments/assets/350357be-41bd-4b3e-9d39-6bd7a6185d26

## Pre-merge author checklist

- [x] I've followed MetaMask Contributor Docs
- [x] I've completed the PR template
- [x] I've documented my code using JSDoc format
- [x] I've applied appropriate labels

## Pre-merge reviewer checklist

- [ ] I've manually tested the PR
- [ ] I confirm this PR addresses all acceptance criteria

### CODE REVIEW

File: apps/storybook-react/.storybook/main.ts [hunk 1]
```diff
@@ -39,6 +39,10 @@ const config: StorybookConfig = {
           __dirname,
           '../../../packages/design-tokens',
         ),
+        '@metamask/design-system-react': path.resolve(
+          __dirname,
+          '../../../packages/design-system-react/src',
+        ),
```
Comment (georgewrmarshall) [inline apps/storybook-react/.storybook/main.ts:45]: Fixes module resolution in Storybook by configuring Vite to use the source files directly from the `src` directory instead of the compiled `dist` files. This ensures all TypeScript types and exports are properly available during development, resolving the "missing TextProps export" error.

Screenshot 2024-11-21 at 5 55 02 PM

File: apps/storybook-react/.storybook/preview.tsx [hunk 2]
```diff
@@ -3,6 +3,7 @@ import '@metamask/design-tokens/src/css/index.css';
 import '../tailwind.css';
 
 import { Preview } from '@storybook/react';
+import { StoryContext, StoryFn } from '@storybook/react';
```
Comment (georgewrmarshall) [inline apps/storybook-react/.storybook/preview.tsx:6]: Fixes types issues in preview file

Screenshot 2024-11-21 at 5 58 09 PM

File: apps/storybook-react/.storybook/preview.tsx [hunk 3]
```diff
@@ -21,18 +22,27 @@ export const globalTypes = {
...
+  const globalColorScheme = context.globals.colorScheme;
 
-  function Wrapper(props) {
+  // Use story-level parameter if available, otherwise fall back to global
+  const colorScheme = storyColorScheme || globalColorScheme;
+
+  function Wrapper({
+    children,
+    ...props
+  }: { children: React.ReactNode } & Record<string, any>) {
```
Comment (georgewrmarshall) [inline apps/storybook-react/.storybook/preview.tsx:35]: Fixing types issues in preview file

Screenshot 2024-11-21 at 5 58 09 PM

File: apps/storybook-react/.storybook/preview.tsx [hunk 4]
```diff
@@ -69,6 +79,18 @@ const preview: Preview = {
...
+        order: [
+          'Design Tokens',
+          'Tailwind Classes',
+          'Components',
+          'Documentation Components',
+          '*', // All other stories
+        ],
+      },
+    },
+    layout: 'fullscreen',
```
Comment (georgewrmarshall) [inline apps/storybook-react/.storybook/preview.tsx:93]: Sets the order of stories

File: apps/storybook-react/stories/Shadows.stories.tsx [hunk 5]
```diff
@@ -24,7 +24,7 @@ const ShadowBox: React.FC<ShadowBoxProps> = ({
 );
 
 const meta = {
-  title: 'Design Tokens/Shadows',
+  title: 'Tailwind Classes/Shadows',
```
Comment (georgewrmarshall) [inline apps/storybook-react/stories/Shadows.stories.tsx:27]: I was considering removing this story as Shadows are covered in the design tokens stories but I think it might be a good place to start creating Tailwind specific recipes and examples. Open to thoughts and ideas on this

File: apps/storybook-react/tailwind.config.js [hunk 6]
```diff
@@ -5,6 +5,7 @@ module.exports = {
   presets: [designSystemPreset],
   content: [
     '../../packages/design-system-react/src/**/*.{js,jsx,ts,tsx}',
+    '../../packages/design-tokens/stories/**/*.{js,jsx,ts,tsx}',
```
Comment (georgewrmarshall) [inline apps/storybook-react/tailwind.config.js:8]: I think we can remove this because we aren't currently using tailwind classnames inside of the design token stories

```suggestion
```

File: packages/design-system-react/src/components/text/Text.tsx [hunk 7]
```diff
@@ -15,6 +15,7 @@ export const Text: React.FC<TextProps> = ({
   ellipsis,
   as,
   color = TextColor.TextDefault,
+  style,
```
Comment (georgewrmarshall) [inline packages/design-system-react/src/components/text/Text.tsx:18]: Move this to it's own PR. Also create an issue to allow spreading of props to the `Text` component

File: packages/design-system-react/tsconfig.json [hunk 8]
```diff
@@ -9,7 +9,10 @@
       "@testing-library/react",
       "@testing-library/jest-dom"
     ],
-    "lib": ["ES2020", "DOM"]
+    "lib": ["ES2020", "DOM"],
+    "module": "ESNext",
+    "moduleResolution": "node",
+    "esModuleInterop": true
```
Comment (georgewrmarshall) [inline packages/design-system-react/tsconfig.json:15]: We can revert this it was a storybook issue not an issue with the design-system-react package phew!
```suggestion
    "lib": ["ES2020", "DOM"],
```

File: packages/design-tokens/stories/BrandColors.stories.tsx [hunk 9]
```diff
@@ -26,9 +26,11 @@ type Story = StoryObj<typeof ColorSwatchGroup>;
 export const Figma: Story = {
   render: () => {
     const { brandColor } = useJsonColor();
-    console.log('brandColor JSON', brandColor);
     return <ColorSwatchGroup swatchData={brandColor} />;
   },
+  parameters: {
+    colorScheme: 'light',
+  },
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/BrandColors.stories.tsx:33]: This locks the Figma story to light mode otherwise we get dual and it doesn't look right.

Screenshot 2024-11-21 at 6 06 21 PM

File: packages/design-tokens/stories/Shadows.stories.tsx [hunk 10]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import type { Meta, StoryObj } from '@storybook/react';
-import { Text } from './components';
+import { Text } from '@metamask/design-system-react';
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/Shadows.stories.tsx:3]: Using design-system-react Text component instead of documentation components

File: packages/design-tokens/stories/ThemeColors.stories.tsx [hunk 11]
```diff
@@ -98,7 +94,6 @@ export const CSSDarkTheme = {
       <div
         style={{
           backgroundColor: 'var(--color-background-default)',
-          margin: '-1rem',
           padding: '1rem',
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/ThemeColors.stories.tsx:97]: Remove this
```suggestion
```

File: apps/storybook-react/stories/Shadows.stories.tsx
Comment (georgewrmarshall) [inline apps/storybook-react/stories/Shadows.stories.tsx:1]: No longer needed we can test tailwind classnames in design token shadows stories

File: packages/design-system-react/src/components/text/Text.constants.ts [hunk 12]
```diff
@@ -16,9 +16,9 @@ export const TEXT_CLASS_MAP: Record<TextVariant, string> = {
   [TextVariant.BodyMd]:
```
Comment (georgewrmarshall) [inline packages/design-system-react/src/components/text/Text.constants.ts:16]: Move to a separate PR

File: packages/design-tokens/stories/Shadows.mdx [hunk 13]
```diff
@@ -13,39 +13,208 @@ Shadows come in four different sizes: XS, SM, MD, and LG. The size of the shadow
 
 <Canvas of={ShadowStories.Size} />
 
-| Size   | JS                | CSS                     |
-| ------ | ----------------- | ----------------------- |
-| **XS** | `shadows.size.xs` | `var(--shadow-size-xs)` |
-| **SM** | `shadows.size.sm` | `var(--shadow-size-sm)` |
-| **MD** | `shadows.size.md` | `var(--shadow-size-md)` |
-| **LG** | `shadows.size.lg` | `var(--shadow-size-lg)` |
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/Shadows.mdx:21]: Unfortunately markdown tables have stopped working in storybook so we have to use jsx

Screenshot 2024-11-22 at 11 57 14 AM

File: packages/design-tokens/stories/Shadows.stories.tsx [hunk 14]
```diff
@@ -1,245 +1,108 @@
...
 import type { Meta, StoryObj } from '@storybook/react';
-import { Text } from './components';
+import { Text, TextColor } from '@metamask/design-system-react';
 import README from './Shadows.mdx';
 
 interface ShadowSwatchProps {
   children: React.ReactNode;
-  style?: React.CSSProperties;
-  size?: 'xs' | 'sm' | 'md' | 'lg';
-  color?: 'default' | 'primary' | 'error';
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/Shadows.stories.tsx:10]: Can use tailwind classnames instead of these

File: packages/design-tokens/stories/Shadows.stories.tsx [hunk 15]
```diff
@@ -1,245 +1,108 @@
...
-          ? `var(--shadow-size-${size}) var(--color-shadow-default)`
-          : `var(--shadow-size-${size}) var(--color-${color}-shadow)`,
-      borderRadius: '4px',
-      display: 'grid',
-      alignContent: 'center',
-      justifyContent: 'center',
-      textAlign: 'center',
-      ...style,
-    }}
+    className={`h-[100px] bg-background-default rounded grid place-content-center text-center ${className}`}
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/Shadows.stories.tsx:16]: Using tailwind instead of inline styles

File: packages/design-tokens/stories/ThemeColors.stories.tsx [hunk 16]
```diff
@@ -89,17 +87,21 @@ export const CSSLightTheme = {
...
+    colorScheme: 'light',
+  },
 };
 
 export const CSSDarkTheme = {
   render: () => {
-    const darkThemeColors = getCSSVariablesFromStylesheet('--color-');
+    console.log('Rendering CSSDarkTheme story');
+    const darkThemeColors = getCSSVariablesFromStylesheet('--color-', 'dark');
+    console.log('Background color:', darkThemeJS.colors.background.default);
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/ThemeColors.stories.tsx:99]: Remove these console logs

File: packages/design-tokens/stories/Typography.mdx [hunk 17]
```diff
@@ -13,9 +13,28 @@ The main font family used in MetaMask products is **Euclid Circular B**
 
 <Canvas of={TypographyStories.FontFamily} />
 
-| Font Family           | JS                   | CSS                                    |
-| --------------------- | -------------------- | -------------------------------------- |
-| **Euclid Circular B** | `Token not exported` | `var(--font-family-euclid-circular-b)` |
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/Typography.mdx:18]: Fixes broken table

File: packages/design-tokens/stories/components/ColorSwatch/ColorSwatch.tsx [hunk 18]
```diff
@@ -1,5 +1,6 @@
 import React, { FunctionComponent } from 'react';
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/components/ColorSwatch/ColorSwatch.tsx:1]: Using `design-system-react` Text component and converting to tailwind css

File: packages/design-tokens/stories/components/Text/Text.stories.tsx
Comment (georgewrmarshall) [inline packages/design-tokens/stories/components/Text/Text.stories.tsx:1]: No longer needed

File: packages/design-tokens/stories/test-utils/getCSSVariablesFromStylesheet.ts [hunk 19]
```diff
@@ -1,7 +1,10 @@
-import fs from 'fs';
-import path from 'path';
-import postcss from 'postcss';
+/* eslint-disable no-restricted-globals */
+/* eslint-disable id-denylist */
+/* eslint-disable prefer-destructuring */
+/* eslint-disable @typescript-eslint/prefer-for-of */
+/* eslint-disable require-unicode-regexp */
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/test-utils/getCSSVariablesFromStylesheet.ts:5]: Need to disable linting for this file as it uses forbidden globals to collect the stylesheet. We can't use the stylesheets from design-tokens because it conflicts with the node environment

File: apps/storybook-react/.storybook/main.ts [hunk 20]
```diff
@@ -30,6 +32,13 @@ const config: StorybookConfig = {
     name: getAbsolutePath('@storybook/react-vite'),
     options: {},
   },
+  docs: {
+    autodocs: true,
+    defaultName: 'Documentation',
+  },
+  core: {
+    builder: '@storybook/builder-vite',
+  },
```
Comment (georgewrmarshall) [inline apps/storybook-react/.storybook/main.ts:41]: Remove
```suggestion
```

File: apps/storybook-react/.storybook/preview.tsx (hunk 2 above)
Comment (georgewrmarshall) [inline apps/storybook-react/.storybook/preview.tsx:6]: Adding types

File: apps/storybook-react/.storybook/preview.tsx [hunk 21]
```diff
@@ -69,6 +79,18 @@ const preview: Preview = {
...
   parameters: {
     controls: { expanded: true },
+    options: {
+      storySort: {
+        order: [
+          'Getting Started',
+          'Design Tokens',
+          'React Components',
+          'Docs Components',
+          '*', // All other stories
```
Comment (georgewrmarshall) [inline apps/storybook-react/.storybook/preview.tsx:89]: Sorting stories

File: apps/storybook-react/README.md [hunk 22]
```diff
@@ -1,17 +1,24 @@
-# `@metamask/storybook-react`
```
Comment (georgewrmarshall) [inline apps/storybook-react/README.md:1]: Updating storybook react readme to accurately describe the packages purpose and serve as an introduction page

Screenshot 2024-11-26 at 1 11 49 PM

File: apps/storybook-react/stories/Introduction.mdx [hunk 23]
```diff
@@ -0,0 +1,7 @@
+import { Meta, Markdown } from '@storybook/blocks';
```
Comment (georgewrmarshall) [inline apps/storybook-react/stories/Introduction.mdx:1]: Renders the README in storybook

File: packages/design-system-react/src/components/Introduction.mdx [hunk 24]
```diff
@@ -0,0 +1,7 @@
+import { Meta, Markdown } from '@storybook/blocks';
```
Comment (georgewrmarshall) [inline packages/design-system-react/src/components/Introduction.mdx:1]: Renders the readme in storybook

File: packages/design-system-react/src/components/button/Button.stories.ts [hunk 25]
```diff
@@ -5,7 +5,7 @@ import { Button } from './Button';
 
 // More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
 const meta: Meta<typeof Button> = {
-  title: 'Components/Button',
+  title: 'React Components/Button',
```
Comment (georgewrmarshall) [inline packages/design-system-react/src/components/button/Button.stories.ts:8]: Updating to React components so the platform is clearer. Eventually we should be able to host the React Native storybook with React Native Components

File: packages/design-tokens/stories/components/ColorSwatch/ColorSwatch.stories.tsx [hunk 26]
```diff
@@ -2,7 +2,7 @@ import type { Meta, StoryObj } from '@storybook/react';
 import { ColorSwatch } from './ColorSwatch';
 
 const meta: Meta<typeof ColorSwatch> = {
-  title: 'Documentation Components/ColorSwatch',
+  title: 'Docs Components/ColorSwatch',
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/components/ColorSwatch/ColorSwatch.stories.tsx:5]: Just shortening name here

File: apps/storybook-react/.storybook/main.ts [hunk 27]
```diff
@@ -12,8 +12,10 @@ function getAbsolutePath(value: string): any {
 }
 const config: StorybookConfig = {
   stories: [
-    '../stories/**/*.mdx',
-    '../stories/**/*.stories.@(js|jsx|mjs|ts|tsx)',
+    '../stories/Introduction.mdx',
+    '../../../packages/design-tokens/stories/Introduction.mdx',
+    '../../../packages/design-tokens/stories/IntroductionColor.mdx',
+    '../../../packages/design-system-react/src/components/Introduction.mdx',
```
Comment (georgewrmarshall) [inline apps/storybook-react/.storybook/main.ts:18]: importing specific mdx files so there is no duplication

Screenshot 2024-11-26 at 1 36 13 PM

File: packages/design-tokens/stories/components/index.tsx [hunk 28]
```diff
@@ -1,3 +1,2 @@
 export { ColorSwatch } from './ColorSwatch';
 export { ColorSwatchGroup } from './ColorSwatchGroup';
-export { Text } from './Text';
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/components/index.tsx:3]: Removing deleted `Text` from index

File: packages/design-tokens/stories/components/ColorSwatchGroup/ColorSwatchGroup.tsx [hunk 29]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import { Theme } from '../../test-utils/useJsonColor';
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/components/ColorSwatchGroup/ColorSwatchGroup.tsx:2]: Converting doc component to use tailwind classnames and the `Text` component from `design-system-react`

File: packages/design-tokens/stories/components/ColorSwatchGroup/ColorSwatchGroup.stories.tsx [hunk 30]
```diff
@@ -2,7 +2,7 @@ import type { Meta, StoryObj } from '@storybook/react';
 import { ColorSwatchGroup } from './ColorSwatchGroup';
 
 const meta: Meta<typeof ColorSwatchGroup> = {
-  title: 'Documentation Components/ColorSwatchGroup',
+  title: 'Docs Components/ColorSwatchGroup',
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/components/ColorSwatchGroup/ColorSwatchGroup.stories.tsx:5]: Just shortening title here

File: packages/design-tokens/stories/Typography.stories.tsx [hunk 31]
```diff
@@ -8,12 +8,12 @@ import { fontFamilies } from '../src/js/typography/fontFamilies';
 import { fontWeights } from '../src/js/typography/fontWeights';
 import { letterSpacing } from '../src/js/typography/letterSpacing';
 
-import { Text } from './components';
+import { Text } from '@metamask/design-system-react';
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/Typography.stories.tsx:11]: Using `Text` component from `design-system-react`

File: packages/design-tokens/stories/ThemeColors.stories.tsx [hunk 32]
```diff
@@ -26,37 +26,35 @@ export const FigmaLightTheme = {
     if (!lightTheme) {
       return null; // or some fallback component
     }
-    console.log('lightTheme', lightTheme);
     return <ColorSwatchGroup swatchData={lightTheme} />;
   },
+  parameters: {
+    colorScheme: 'light',
+  },
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/ThemeColors.stories.tsx:33]: We want to lock this story to light mode

File: packages/design-tokens/stories/ThemeColors.stories.tsx [hunk 33]
```diff
@@ -26,37 +26,35 @@ export const FigmaLightTheme = {
...
+      darkTheme.background !== null &&
+      'default' in darkTheme.background &&
+      typeof darkTheme.background.default === 'object' &&
+      darkTheme.background.default !== null &&
+      'value' in darkTheme.background.default
+        ? darkTheme.background.default.value
+        : undefined;
+
+    return <ColorSwatchGroup swatchData={darkTheme} theme={backgroundColor} />;
   },
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/ThemeColors.stories.tsx:55]: Don't love how hard this is to read, but it fixes a type issue, and since this is just for Storybook and not an actual component, I think it’s fine for now.  

Screenshot of the code fixing a type issue

File: packages/design-tokens/stories/ThemeColors.stories.tsx [hunk 34]
```diff
@@ -26,37 +26,35 @@ export const FigmaLightTheme = {
...
   parameters: {
-    backgrounds: {
-      default: 'dark',
-      values: [{ name: 'dark', value: brandColor.grey[800].value }],
-    },
+    colorScheme: 'dark',
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/ThemeColors.stories.tsx:57]: Locking story to dark mode

File: packages/design-tokens/stories/ThemeColors.stories.tsx [hunk 35]
```diff
@@ -89,17 +87,21 @@ export const CSSLightTheme = {
       </div>
     );
   },
+  parameters: {
+    colorScheme: 'light',
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/ThemeColors.stories.tsx:91]: Locking story to light mode

File: packages/design-tokens/stories/ThemeColors.stories.tsx [hunk 36]
```diff
@@ -89,17 +87,21 @@ export const CSSLightTheme = {
...
+    console.log('Rendering CSSDarkTheme story');
+    const darkThemeColors = getCSSVariablesFromStylesheet('--color-', 'dark');
+    console.log('Background color:', darkThemeJS.colors.background.default);
+
     return (
       <div
         style={{
           backgroundColor: 'var(--color-background-default)',
-          margin: '-1rem',
-          padding: '1rem',
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/ThemeColors.stories.tsx:102]: Not needed we have the storybook wrapper

File: packages/design-tokens/package.json [hunk 37]
```diff
@@ -51,6 +51,7 @@
   },
   "devDependencies": {
     "@metamask/auto-changelog": "^3.4.4",
+    "@metamask/design-system-react": "workspace:^",
```
Comment (georgewrmarshall) [inline packages/design-tokens/package.json:54]: We are using the `design-system-react` `Text` component in stories instead of duplicating components

File: apps/storybook-react/README.md (hunk 22 above)
Comment (georgewrmarshall) [inline apps/storybook-react/README.md:1]: Updating storybook readme to serve as a better description of what this package is for and the introduction page for storybook

Screenshot 2024-11-27 at 10 47 39 AM

File: apps/storybook-react/.storybook/preview.tsx [hunk 38]
```diff
@@ -69,6 +79,18 @@ const preview: Preview = {
...
+        ],
+      },
+    },
+    layout: 'fullscreen', // removes default padding around stories
```
Comment (georgewrmarshall) [inline apps/storybook-react/.storybook/preview.tsx:93]: We are including padding at the `Wrapper` on line `41`

File: packages/design-tokens/stories/ThemeColors.stories.tsx (hunk 33 above)
Comment (brianacnguyen) [reply packages/design-tokens/stories/ThemeColors.stories.tsx:55]: holy conditioning

**Review by brianacnguyen — APPROVED** [review]
